src/pages/landing/movies/MovieDetailPage.vue

```diff
@@ -0,0 +1,67 @@
+<script>
+import { defineComponent } from 'vue';
+import { getMovieData } from '@/services/api.js';
+import { mapState } from 'pinia';
+import { useMainStore } from '@/store/index.js';
+
+import TheBreadcrumb from '@/components/TheBreadcrumb.vue';
+import MovieDetail from '@/components/movies/MovieDetail.vue';
+
+export default defineComponent({
+  components: {
+    MovieDetail,
+    TheBreadcrumb,
+  },
+  props: {
+    movieId: {
+      type: String,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      isLoading: true,
+      storedMovie: null,
+      movieTitle: null,
```

Comment:
Initially, `storedMovie` variable is null so accessing `this.storedMovie.title` synchronously isn't really an option. This would throw an error because I try to pass it as a prop to `<TheBreadcrumb :second-item="storedMovie.title">`. But accessing `this.movieTitle` works - it's `null`, but at least it exists, so it works as a placeholder variable waiting to be changed when the data is finally fetched.

Anyway, thanks to you I found something else here: `movieTitle` should be initially an empty string, not a `null`, as the prop `second-item` expects the `type` of a `String`. I saw no warnings from Vite about that though 🤔

Is it fine to use this variable like that, or should I treat this scenario differently?

Reply:
You could have a check in your template to see if the movie is there yet or not (we can look at the `isLoading` flag).
But if you want to, you could refactor this one to... a computed - it could return an empty string if there is no movie or the tile if it is there 😄

Reply:
Well I didn't want to use the flag on `<TheBreadcrumb />` since in my opinion it should not disappear from the screen while one of its inner elements is being loaded. Also I rather not pass that flag into it, making the component more complicated.

I'm aware of that it is not really the best solution in every scenario, but I'll go with the computed property 👍